mayastor/src/bdev/nexus/nexus_child_status_config.rs

```diff
@@ -0,0 +1,139 @@
+//! The purpose of this module is to persist child status information across
+//! Mayastor restarts.
+//!
+//! The load function should only be called when Mayastor is initialising. This
+//! will read in the child status information from the config file and use it to
+//! initialise the in-memory ChildStatusConfig structure. The apply function can
+//! then be called which will set the status of all children to match the
+//! configuration.
+//!
+//! The save function should be called whenever a child's status is updated.
+//! This will update the configuration file but WILL NOT update the in-memory
+//! ChildStatusConfig structure as this is only required on startup and not
+//! during runtime.
+
+use crate::bdev::nexus::{
+    instances,
+    nexus_channel::DREvent,
+    nexus_child::{NexusChild, StatusReasons},
+};
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fs, fs::File, io::Write};
+
+type ChildName = String;
+pub const CONFIG_FILE: &str = "/tmp/child_status.yaml";
```

Comment:
might be worth considering getting this a CLI parameter? We do the same with the other config file.

Reply:
Done.